### PR TITLE
Added MATE support for media keys

### LINKED ISSUE #915: Media keys do nothing MATE
**ENV:**
Fedora 23 MATE spin
MATE 1.12.1
Player verison 3.1.0
**Issue Descriptions:**
The media keys, they do nothing.   I don't believe I've got Chrome hogging them.

I was able to work around this issue by mapping the keys to dbus methods.
ex.
`dbus-send --session --type=method_call --dest=org.mpris.MediaPlayer2.google-play-music-desktop-player /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player.PlayPause
`

### PR DESCRIPTION
**Pull request by mattcfox**
This should fix #915 
Unfortunately this fix is somewhat theoretical because I seem to have some difficulty building on my setup.

### CODE REVIEW

**Comment by jostrander** [comment]
I'll have to spin up a vm with mate and test this out, ideally I'd like to avoid the duplication of code as much as possible, so might extract the callback to a function.

File: src/main/features/linux/mediaKeysDBus.js [hunk 1]
```diff
@@ -19,6 +19,21 @@ try {
       iface.GrabMediaPlayerKeys(0, 'org.gnome.SettingsDaemon.MediaKeys'); // eslint-disable-line
     }
   });
+  session.getInterface('org.mate.SettingsDaemon', '/org/mate/SettingsDaemon/MediaKeys',
```
Comment (MarshallOfSound) [inline src/main/features/linux/mediaKeysDBus.js]: This is exact code duplication with two strings different.  Can you extract it to another function and pass in a single parameter (`gnome` and `mate`)

**Comment by jostrander** [comment]
Also, once this builds on CI you can grab the deb and install it to verify it works, you may have to remove your current install first however.

File: src/main/features/linux/mediaKeysDBus.js (hunk 1 above)
Comment (mattcfox) [inline src/main/features/linux/mediaKeysDBus.js]: Sure, was being a bit quick and sloppy

File: src/main/features/linux/mediaKeysDBus.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import DBus from 'dbus';
 
-function registerBindings(desktopEnv, session){
-  session.getInterface('org.'+desktopEnv+'.SettingsDaemon', '/org/'+desktopEnv+'/SettingsDaemon/MediaKeys','org.'+desktopEnv+'.SettingsDaemon.MediaKeys', (err, iface) => {
+function registerBindings(desktopEnv, session) {
+  session.getInterface('org.' + desktopEnv + '.SettingsDaemon', '/org/' + desktopEnv + '/SettingsDaemon/MediaKeys',
```
Comment (jostrander) [inline src/main/features/linux/mediaKeysDBus.js]: Surprised lint didn't pick this up, but these should use the es6 syntax:
```
`org.${desktopEnv}.SettingsDaemon`
```

**Comment by MarshallOfSound** [comment]
You have a few linting errors

https://travis-ci.org/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/builds/121589427

File: src/main/features/linux/mediaKeysDBus.js [hunk 3]
```diff
@@ -1,11 +1,8 @@
 import DBus from 'dbus';
 
-try {
-  const dbus = new DBus();
-  const session = dbus.getBus('session');
-
-  session.getInterface('org.gnome.SettingsDaemon', '/org/gnome/SettingsDaemon/MediaKeys',
-  'org.gnome.SettingsDaemon.MediaKeys', (err, iface) => {
+function registerBindings(desktopEnv, session) {
+  session.getInterface('org.' + desktopEnv + '.SettingsDaemon', '/org/' + desktopEnv + '/SettingsDaemon/MediaKeys',
```
Comment (MarshallOfSound) [inline src/main/features/linux/mediaKeysDBus.js]: This should be in a style like

<pre>`org.${desktopEnv}.SettingsDaemon`</pre>

**Comment by mattcfox** [comment]
Yea, apologies for those, I don't do a ton of JS stuff.

**Comment by mattcfox** [comment]
@jostrander, I think I can grab the artifacts from circleci.  If you want to get me the RPM, I'll try it on my setup.

**Comment by mattcfox** [comment]
@jostrander, I don't think I can grab the artifacts from circleci.  If you want to grab me the RPM, I can try it on my setup.

**Comment by jostrander** [comment]
32 or 64?

**Comment by mattcfox** [comment]
64

**Comment by jostrander** [comment]
https://circle-artifacts.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/342/artifacts/0/home/ubuntu/Google-Play-Music-Desktop-Player-UNOFFICIAL-/dist/installers/fedora/google-play-music-desktop-player-3.1.0.amd64.rpm

**Comment by mattcfox** [comment]
That fixed it

**Comment by jostrander** [comment]
I'll grab a deb and make sure its still working for me, one second...

**Comment by jostrander** [comment]
All good to merge if CS is good @MarshallOfSound :+1: Thanks @mattcfox for the quick work!

**Comment by mattcfox** [comment]
No problem :smile:

**Pull request merged**